src/flint/fq_nmod.jl

```diff
@@ -49,6 +49,44 @@ function coeff(x::fq_nmod, n::Int)
                 (Ref{fq_nmod}, Int), x, n)
 end
 
+function coeffs_raw(x::fq_nmod)
+   @GC.preserve x begin
+      len = degree(parent(x))
+      V = unsafe_wrap(Vector{UInt}, reinterpret(Ptr{UInt}, x.coeffs), x.length)
+      Vcopy = Vector{UInt}(undef, len)
+      for i = 1:x.length
+         Vcopy[i] = V[i]
+      end
+      for i = x.length + 1:len
+         Vcopy[i] = 0
+      end
+   end
+   return Vcopy
+end
+
+function setindex_raw!(x::fq_nmod, c::UInt, i::Int)
+   len = degree(parent(x))
+   V = unsafe_wrap(Vector{UInt}, reinterpret(Ptr{UInt}, x.coeffs), len)
+   if i + 1 > x.length
+      if !iszero(c)
+         for j = x.length + 1:i
+            V[j] = 0
+         end
+         V[i + 1] = c
+         x.length = i + 1
```

Comment:
I don't think we can just change `x.length` here. This is a memory corruption.

Reply:
However, at the end of the day, would it just be better to pun the fq_nmod_struct into a nmod_poly_struct and call the correct nmod_poly function? Why write the same code twice in julia?

Reply:
This type is supposed to always allocate deg(R) coefficients when it is initialised, so it shouldn't be a corruption. A lot of code would be broken if that weren't the case.

Reply:
> This type is supposed to always allocate deg(R) coefficients when it is initialised,
This unfortunately is of no consequence. The fq_nmod you get back after calling a flint function on an fq_nmod can have any alloc even if it went in with at least deg(R) alloc. Code like this in nemo is just asking for trouble.
I would pun it to a nmod_poly, call the correct nmod_poly function, and avoid the incorrect code duplication.

Reply:
I will see if I can pun it in Nemo.